include/alpaka/event/EventUniformCudaHipRt.hpp

```diff
@@ -191,43 +232,51 @@ namespace alpaka
         {
             //#############################################################################
-            //! The CUDA RT queue enqueue trait specialization.
+            //! The CUDA/HIP RT queue enqueue trait specialization.
             template<>
             struct Enqueue<
-                queue::QueueCudaRtNonBlocking,
-                event::EventCudaRt>
+                queue::QueueUniformCudaHipRtNonBlocking,
+                event::EventUniformCudaHipRt>
             {
                 //-----------------------------------------------------------------------------
                 ALPAKA_FN_HOST static auto enqueue(
-                    queue::QueueCudaRtNonBlocking & queue,
-                    event::EventCudaRt & event)
+                    queue::QueueUniformCudaHipRtNonBlocking & queue,
+                    event::EventUniformCudaHipRt & event)
                 -> void
                 {
                     ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
-
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
                     ALPAKA_CUDA_RT_CHECK(cudaEventRecord(
-                        event.m_spEventImpl->m_CudaEvent,
-                        queue.m_spQueueImpl->m_CudaQueue));
+                        event.m_spEventImpl->m_UniformCudaHipEvent,
+                        queue.m_spQueueImpl->m_UniformCudaHipQueue));
+#else
+                    ALPAKA_HIP_RT_CHECK(hipEventRecord(
+                         event.m_spEventImpl->m_UniformCudaHipEvent,
+                         queue.m_spQueueImpl->m_UniformCudaHipQueue));
+#endif
                 }
             };
             //#############################################################################
-            //! The CUDA RT queue enqueue trait specialization.
+            //! The CUDA/HIP RT queue enqueue trait specialization.
             template<>
             struct Enqueue<
-                queue::QueueCudaRtBlocking,
-                event::EventCudaRt>
+                queue::QueueUniformCudaHipRtBlocking,
+                event::EventUniformCudaHipRt>
             {
                 //-----------------------------------------------------------------------------
                 ALPAKA_FN_HOST static auto enqueue(
-                    queue::QueueCudaRtBlocking & queue,
-                    event::EventCudaRt & event)
+                    queue::QueueUniformCudaHipRtBlocking & queue,
+                    event::EventUniformCudaHipRt & event)
                 -> void
                 {
                     ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
-
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
                     ALPAKA_CUDA_RT_CHECK(cudaEventRecord(
-                        event.m_spEventImpl->m_CudaEvent,
-                        queue.m_spQueueImpl->m_CudaQueue));
+#else
+                    ALPAKA_HIP_RT_CHECK(hipEventRecord(
+#endif
+                        event.m_spEventImpl->m_UniformCudaHipEvent,
+                        queue.m_spQueueImpl->m_UniformCudaHipQueue));
```

Comment:
clang says that this is undefined behaviour:
`error: embedding a directive within macro arguments has undefined behavior [-Werror,-Wembedded-directive]`